src/decision/src/VisionDecision.cpp

```diff
@@ -24,13 +30,321 @@ VisionDecision::VisionDecision(int argc, char **argv, std::string node_name) {
     uint32_t queue_size = 10;
     twist_publisher = nh.advertise<geometry_msgs::Twist>(twist_topic, queue_size);
 
+
 }
 
 // This is called whenever a new message is received
-void VisionDecision::imageCallBack(const sensor_msgs::Image::ConstPtr& raw_scan) {
+void VisionDecision::imageCallBack(const sensor_msgs::Image::ConstPtr& image_scan) {
     // Deal with new messages here
+    geometry_msgs::Twist twistMsg;
+
+    // Decide how much to turn
+    int relativeAngle = getDesiredAngle(image_scan->height/2.0, image_scan);
+
+    // Initialize linear velocities to 0
+    twistMsg.linear.x = getDesiredSpeed(relativeAngle);
+    twistMsg.linear.y = 0;
+    twistMsg.linear.z = 0;
+
+    // Initialize x and y angular velocities to 0
+    twistMsg.angular.x = 0;
+    twistMsg.angular.y = 0;
+
+    // Decide how fast to turn
+    twistMsg.angular.z = getDesiredAngularSpeed(relativeAngle);
+
+    // Do the turn
+    double t0 = ros::Time::now().toSec();
+    double current_angle;
+    ros::Rate loop_rate(1000);
+    do{
+        publishTwist(twistMsg);
+        double t1 = ros::Time::now().toSec();
+        current_angle = twistMsg.angular.z * (t1-t0);
+        ros::spinOnce();
+        loop_rate.sleep();
+    }while(current_angle<abs(relativeAngle));
```

Comment:
The twist message should just be published once, as the main decision node will use the last received twist message until a new one is received. If a new twist message is published, it should be because we got a new image. Also, this presumes that the robot will turn at exactly the rate specified in the twist message, which it probably won't. This is one of the strengths of reactionary systems. Even if our numbers (turn rate, linear speed) are off a bit, since it's directly reacting to what actually happens in the real world, we can still get decent results.